### PR TITLE
Add status details support to server HTTP handler

### PR DESCRIPTION
**Pull request by c4milo**
Complementing the work done in https://github.com/grpc/grpc-go/pull/1156/files

### CODE REVIEW

File: transport/handler_server.go [hunk 1]
```diff
@@ -186,7 +187,14 @@ func (ht *serverHandlerTransport) WriteStatus(s *Stream, st *status.Status) erro
 			h.Set("Grpc-Message", encodeGrpcMessage(m))
 		}
 
-		// TODO: Support Grpc-Status-Details-Bin
+		if p := st.Proto(); p != nil && len(p.Details) > 0 {
+			stBytes, err := proto.Marshal(p)
+			if err != nil {
+				panic(err)
```
Comment (dfawley) [inline transport/handler_server.go]: `// TODO: return an error instead.` please, to match `http2_server.go`.

We should never panic unless it's unrecoverable.

Comment (c4milo) [inline transport/handler_server.go]: Sure, I missed that one. I'll fix it as soon as I can get my hands on my laptop.

Comment (c4milo) [inline transport/handler_server.go]: @dfawley done

**Comment by dfawley** [comment]
LGTM, thank you for the PR!

**Comment by c4milo** [comment]
@dfawley np, thanks for taking the time to review!

**Pull request merged**